### ISSUE TITLE
No variable assignment in for-loop. Directly renders value into output.

### ISSUE THREAD

**Issue by Donnerbalken** [lunet-io/scriban#105]
This one might be a follow up on #97 
I experienced the issue at the same time I found #97. So I thought these might be related.
After installing 1.2.9 #97 has been solved, but this one still occurs.

```
{{~ 
    for position in positions 
        # this line directly outputs into the template, but does not assign the variable
        netto = position.netto | format_number
       # this line works and assigns the value to the variable
        brutto = position.brutto | math.format "0.00" | string.replace "." ""

~}}
( ):1100;{{ netto }};;;3114;0;0;"FA0000002446";"608";2018-10-31;75113;;;"company";;;;;15;;;;;;211;;
lala {{ position.account }} stuff
   
{{~ end ~}}
{{
func format_number
    ret $0 | math.format "0.00" | string.replace "." ""
end
}}
```

Result:
150510( ):1100;;;;3114;0;0;"FA0000002446";"608";2018-10-31;75113;;;"company";;;;;15;;;;;;211;;

The output of "150510" at the very beginning of the line is wrong. But the replacement of the placeholder after 1100; does not happen.

**Comment by xoofx** [comment]
Could you try latest preview `2.0.0+`, I'm not sure it is related to #97 but to pipeline functions

**Comment by Donnerbalken** [comment]
Same behavior with version 2.0.0-alpha-004

**Comment by xoofx** [comment]
Ok, that might be different then, I will have a look

**Comment by Donnerbalken** [comment]
I also tested `netto = format_number position.netto`
Not sure if it is handled differently, but still the same output

**Comment by xoofx** [comment]
The function `format_number` should be declared before using it as per example in commit 1b2fa67
You have also likely disabled exception on unresolved symbols, because you should have had an exception.

**Comment by xoofx** [comment]
This should be fixed by commit 20b4054 and available shortly in `2.0.0-alpha-005`
